x-pack/plugins/security_solution/public/management/components/paginated_content/paginated_content.tsx

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, {
+  ComponentProps,
+  ComponentType,
+  FunctionComponent,
+  Key,
+  memo,
+  ReactElement,
+  ReactNode,
+  useCallback,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  EuiEmptyPrompt,
+  EuiIcon,
+  EuiProgress,
+  EuiSpacer,
+  EuiTablePagination,
+  EuiTablePaginationProps,
+  EuiText,
+  Pagination,
+} from '@elastic/eui';
+import styled from 'styled-components';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { v4 as generateUUI } from 'uuid';
+import { useTestIdGenerator } from '../hooks/use_test_id_generator';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type ComponentWithAnyProps = ComponentType<any>;
+
+export interface PaginatedContentProps<T, C extends ComponentWithAnyProps> {
```

Comment:
This component will allow us to reuse a lot of code between trusted apps, event filters and upcoming listings 👍  😃